roc/config.go

```diff
@@ -5,68 +5,238 @@ package roc
+	//
+	// Allowed operations:
+	//  - bind    (sender, receiver)
+	//  - connect (sender, receiver)
+	//
+	// Allowed protocols:
+	//  - ProtoRtsp
+	///
+	InterfaceConsolidated Interface = 1
+
+	// Interface for audio stream source data.
+	//
+	// Allowed operations:
+	//  - bind    (receiver)
+	//  - connect (sender)
+	//
+	// Allowed protocols:
+	//  -  ProtoRtp
+	//  -  ProtoRtpRs8mSource
+	//  -  ProtoRtpLdpcSource
+	///
+	InterfaceAudioSource Interface = 11
+
+	// Interface for audio stream repair data.
+	//
+	// Allowed operations:
+	//  - bind    (receiver)
+	//  - connect (sender)
+	//
+	// Allowed protocols:
+	//  -  ProtoRs8mRepair
+	//  -  ProtoLdpcRepair
+	///
+	InterfaceAudioRepair Interface = 12
+
+	// Interface for audio control messages.
+	//
+	// Allowed operations:
+	//  - bind    (sender, receiver)
+	//  - connect (sender, receiver)
+	//
+	// Allowed protocols:
+	//  -  ProtoRrcp
+	///
+	InterfaceAudioControl Interface = 13
+)
+
+// Network protocol.
+// Defines URI scheme of Endpoint.
 type Protocol int
 
 const (
-	// Bare RTP (RFC 3550).
-	ProtoRtp Protocol = 1
+	// RTSP 1.0 (RFC 2326) or RTSP 2.0 (RFC 7826).
+	//
+	// Interfaces:
+	//  - InterfaceConsolidated
+	//
+	// Transports:
+	//   - for signaling: TCP
+	//   - for media: RTP and RTCP over UDP or TCP
+	///
+	ProtoRtsp Protocol = 10
+
+	// RTP over UDP (RFC 3550).
+	//
+	// Interfaces:
+	//  - InterfaceAudioSource
+	//
+	// Transports:
+	//  - UDP
+	//
+	// Audio encodings:
+	//   - PacketEncodingAvpL16
+	//
+	// FEC encodings:
+	//   - none
+	///
+	ProtoRtp Protocol = 20
 
 	// RTP source packet (RFC 3550) + FECFRAME Reed-Solomon footer (RFC 6865) with m=8.
-	ProtoRtpRs8mSource Protocol = 2
+	//
+	// Interfaces:
+	//  - InterfaceAudioSource
+	//
+	// Transports:
+	//  - UDP
+	//
+	// Audio encodings:
+	//  - similar to ProtoRtp
+	//
+	// FEC encodings:
+	//  - FecEncodingRs8m
+	///
+	ProtoRtpRs8mSource Protocol = 30
 
 	// FEC repair packet + FECFRAME Reed-Solomon header (RFC 6865) with m=8.
-	ProtoRs8mRepair Protocol = 3
+	//
+	// Interfaces:
+	//  - InterfaceAudioRepair
+	//
+	// Transports:
+	//  - UDP
+	//
+	// FEC encodings:
+	//  - FecEncodingRs8m
+	///
+	ProtoRs8mRepair Protocol = 31
 
 	// RTP source packet (RFC 3550) + FECFRAME LDPC-Staircase footer (RFC 6816).
-	ProtoRtpLdpcSource Protocol = 4
+	//
+	// Interfaces:
+	//  - InterfaceAudioSource
+	//
+	// Transports:
+	//  - UDP
+	//
+	// Audio encodings:
+	//  - similar to ProtoRtp
+	//
+	// FEC encodings:
+	//  - FecEncodingLdpcStaircase
+	///
+	ProtoRtpLdpcSource Protocol = 32
 
 	// FEC repair packet + FECFRAME LDPC-Staircase header (RFC 6816).
-	ProtoLdpcRepair Protocol = 5
+	//
+	// Interfaces:
+	//  - InterfaceAudioRepair
+	//
+	// Transports:
+	//  - UDP
+	//
+	// FEC encodings:
+	//  - FecEncodingLdpcStaircase
+	///
+	ProtoLdpcRepair Protocol = 33
+
+	// RTCP over UDP (RFC 3550).
+	//
+	// Interfaces:
+	//  - InterfaceAudioControl
+	//
+	// Transports:
+	//  - UDP
+	///
+	ProtoRrcp Protocol = 70
```

Comment:
just a question - are these consts taken from the c code? I'm wondering what's the logic behind the scattered values

Reply:
1. Yep, numbers come from C API.

2. They are scattered because I think new values will be added in between, and it would be nice to keep their numeric values grouped. For example, aggregate protocols will be 10-20, transport protocols will be 20-30, FEC source and repair protocols start from 30, etc.

3. In future it would be nice to create a script that either automatically updates config.go file from C header or maybe just verifies that they're in-sync and reports what is missing / different. I don't mean a generic binding generator, just some simple solution specific to this given file. Let me know if you have ideas how we could do it...

Reply:
Another thing that bothers me is that all these comments can quickly become outdated compared to the C API. Not sure what we could do here.